plugin.py

```diff
@@ -37,91 +50,60 @@ def lsp_expand_variables(window, var):
-        )
-
-
-def onCommandDone():
-    logAndShowMessage('LSP-intelephense: Server installed.')
-
-
-def runCommand(onExit, popenArgs):
-    """
-    Runs the given args in a subprocess.Popen, and then calls the function
-    onExit when the subprocess completes.
-    onExit is a callable object, and popenArgs is a list/tuple of args that
-    would give to subprocess.Popen.
-    """
-    def runInThread(onExit, popenArgs):
-        try:
-            if sublime.platform() == 'windows':
-                subprocess.check_call(popenArgs, shell=True)
-            else:
-                subprocess.check_call(popenArgs)
-            onExit()
-        except subprocess.CalledProcessError as error:
-            logAndShowMessage('LSP-intelephense: Error while installing the server.', error)
-        return
-    thread = threading.Thread(target=runInThread, args=(onExit, popenArgs))
-    thread.start()
-    # returns immediately after the thread starts
-    return thread
-
-
 def is_node_installed():
-    return shutil.which('node') is not None
-
-
-def logAndShowMessage(msg, additional_logs=None):
-    print(msg, '\n', additional_logs) if additional_logs else print(msg)
-    sublime.active_window().status_message(msg)
+    return shutil.which("node") is not None
 
 
 class LspIntelephensePlugin(LanguageHandler):
     @property
     def name(self) -> str:
-        return 'lsp-intelephense'
+        return PACKAGE_NAME.lower()
 
     @property
     def config(self) -> ClientConfig:
-        settings = sublime.load_settings("LSP-intelephense.sublime-settings")
-        client_configuration = settings.get('client')
+        # Calling setup() also here as this might run before `plugin_loaded`.
+        # Will be a no-op if already ran.
+        # See https://github.com/sublimelsp/LSP/issues/899
+        server.setup()
+
+        configuration = self.migrate_and_read_configuration()
+
         default_configuration = {
-            "command": [
-                'node',
-                server_path,
-                '--stdio'
-            ],
-            "languages": [
-                {
-                    "languageId": "php",
-                    "scopes": ["source.php"],
-                    "syntaxes": ["Packages/PHP/PHP.sublime-syntax"]
-                }
-            ]
+            "enabled": True,
+            "command": ["node", server.binary_path, "--stdio"],
         }
-        default_configuration.update(client_configuration)
-        default_configuration['initializationOptions'] = lsp_expand_variables(sublime.active_window(), client_configuration['initializationOptions'])
-        return read_client_config('lsp-intelephense', default_configuration)
+
+        default_configuration.update(configuration)
+        default_configuration["initializationOptions"] = lsp_expand_variables(
+            sublime.active_window(), default_configuration.get("initializationOptions", {})
```

Comment:
Hmm, `sublime.active_window()` sucks. Ideally would pass the actual window here.
(not a new issue of course)